simplegrid/dqn_agent.py

```diff
@@ -35,27 +35,28 @@ def remember(self, state, action, reward, next_state, done):
 
     # get action
     def act(self, state):
-
-        # select random action with prob=epsilon else action=maxQ
+        '''
+        state is a 2-dimensional ndarray representing the surrounding squares
+        returns a random action with prob=epsilon else action=maxQ
+        '''
         if np.random.rand() <= self.epsilon:
             return random.randrange(self.action_size)
-        state = np.reshape(state, (1, -1))
         act_values = self.predict(state)
-        return np.argmax(act_values[0])  # returns action
+        return np.argmax(act_values)  # returns action
```

Comment:
Hmm, waarom halen we hier de [0] weg? Het maakt niet echt uit natuurlijk, maar in principe hebben we hier 1 batch met waardes, dus [0] vind ik leesbaarder?

Reply:
Mee eens. act_values heeft shape (1,4) dus act_values[0] is netter.